pkg/virt-controller/watch/vmi.go

```diff
@@ -595,9 +599,13 @@ func (c *VMIController) updateStatus(vmi *virtv1.VirtualMachineInstance, pod *k8
 		// Nothing here
 		break
 	default:
-		return fmt.Errorf("unknown vmi phase %v", vmi.Status.Phase)
+		return nil, nil, fmt.Errorf("unknown vmi phase %v", vmi.Status.Phase)
 	}
 
+	return vmiCopy, syncErr, nil
+}
+
+func (c *VMIController) modifyVMIOnAPIServer(vmi *virtv1.VirtualMachineInstance, vmiCopy *virtv1.VirtualMachineInstance, syncErr syncError) error {
```

Comment:
what about the section where we are updating the vmi? shouldn't we also deal with it here?

Reply:
It is handled here.